build.gradle

```diff
@@ -585,6 +585,7 @@ dependencies {
     // used for serializing/deserializing rcf models.
     compile group: 'io.protostuff', name: 'protostuff-core', version: '1.7.4'
     compile group: 'io.protostuff', name: 'protostuff-runtime', version: '1.7.4'
+    compileOnly group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
```

Comment:
I have been trying to add this successfully on my own branch and I kept getting "jar hell" errors. Do you know how adding it as compileOnly helped solved this if you also encountered such errors at first?

Reply:
`commons-lang3` was introduced by job-scheduler-spi in 1.2.4 which using `complieOnly`, check line 566.

If change to `compile`, it will cause jar hell in integTest
```
> Task :integTest FAILED
Exec output and error:
| Output for ./bin/opensearch-plugin:-> Installing file:/local/home/ylwu/code/os/anomaly-detection-2/src/test/resources/job-scheduler/opensearch-job-scheduler-1.3.0.0-SNAPSHOT.zip
| -> Downloading file:/local/home/ylwu/code/os/anomaly-detection-2/src/test/resources/job-scheduler/opensearch-job-scheduler-1.3.0.0-SNAPSHOT.zip
| -> Installed opensearch-job-scheduler with folder name opensearch-job-scheduler
| -> Installing file:/local/home/ylwu/code/os/anomaly-detection-2/build/distributions/opensearch-anomaly-detection-1.3.0.0-SNAPSHOT.zip
| -> Downloading file:/local/home/ylwu/code/os/anomaly-detection-2/build/distributions/opensearch-anomaly-detection-1.3.0.0-SNAPSHOT.zip
| -> Failed installing file:/local/home/ylwu/code/os/anomaly-detection-2/build/distributions/opensearch-anomaly-detection-1.3.0.0-SNAPSHOT.zip
| -> Rolling back opensearch-job-scheduler
| -> Rolled back opensearch-job-scheduler
| -> Rolling back file:/local/home/ylwu/code/os/anomaly-detection-2/build/distributions/opensearch-anomaly-detection-1.3.0.0-SNAPSHOT.zip
| -> Rolled back file:/local/home/ylwu/code/os/anomaly-detection-2/build/distributions/opensearch-anomaly-detection-1.3.0.0-SNAPSHOT.zip
| Exception in thread "main" java.lang.IllegalStateException: failed to load plugin opensearch-anomaly-detection due to jar hell
|       at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:678)
|       at org.opensearch.plugins.InstallPluginCommand.jarHellCheck(InstallPluginCommand.java:845)
|       at org.opensearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:813)
|       at org.opensearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:858)
|       at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:263)
|       at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:237)
|       at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:100)
|       at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
|       at org.opensearch.cli.MultiCommand.execute(MultiCommand.java:104)
|       at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
|       at org.opensearch.cli.Command.main(Command.java:101)
|       at org.opensearch.plugins.PluginCli.main(PluginCli.java:60)
| Caused by: java.lang.IllegalStateException: jar hell!
| class: org.apache.commons.lang3.AnnotationUtils$1
| jar1: /local/home/ylwu/code/os/anomaly-detection-2/build/testclusters/integTest-0/distro/1.3.0-ARCHIVE/plugins/opensearch-job-scheduler/opensearch-job-scheduler-spi-1.3.0.0-SNAPSHOT.jar
| jar2: /local/home/ylwu/code/os/anomaly-detection-2/build/testclusters/integTest-0/distro/1.3.0-ARCHIVE/plugins/.installing-1830238384624697112/commons-lang3-3.12.0.jar
|       at org.opensearch.bootstrap.JarHell.checkClass(JarHell.java:317)
|       at org.opensearch.bootstrap.JarHell.checkJarHell(JarHell.java:212)
|       at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:664)
|       ... 11 more
```